### DISCUSSION TITLE
Can I convert  .caf Files?

### DISCUSSION THREAD

**Discussion by Meeeeow** [Markemp/Cryengine-Converter#120] [General]
Hello, I unpacked ArcheAge and used the program to convert objects into .dae. However, I wonder if Animations' .caf can also be converted.
It can be awkward using a translator. Thank you.

**Comment by Markemp** [comment]
Animations are not currently supported, but they are something I am interested in adding in the future.  Unfortunately I don't have a time frame on when that will be though.

**Comment by Dizcordum** [comment]
Im also interested in animations convertion
According to cryengine docs, there is a pretty simple pipeline conversion
that goes from any standart animation file like fbx or even dae, then in modelling programm like maya it exports with .caf extension
also, there is a tool in cryengine itself, that allows to load skinned mesh with animation (.fbx) and it automaticly converts animation from it to .caf format
When there will be update on this?)

**Comment by Kjasi** [comment]
The decoded pipeline is simple, but reading the compressed animation data isn't. Especially when trying to support multiple game's formats, multiple compression types, and multiple animation sources. It's not all boned .caf files; .chr can house animations as well, animating whole models...

And that's not even discussing the topic of converting translations, and rotation matrices... Because rotating a bone 90 degrees in CryEngine doesn't necessarily rotate the same way when put into a .dae file...

**Reply by Dizcordum** [reply]
i found that caf and cgf files starts from same headers and contains same bone data
but also cgf contains mesh and material info and caf not
and caf contains bone translation and rotation data and cgf not
as i said earlier, if you import fbx mesh with animation in cryengine, it separates the animation and converts it to caf format
So its kinda doesnt matter what where used on input when caf output is always superior and universal

**Comment by Dizcordum** [comment]
I did also found in docs that .caf (Compressed Animation File) is stripped animation file that still has all the useful data abot translation and rotation to the be used in game 
and that compression is defined in .dba file. That file contents all the info about compression filter, tags etc for animations so the converted .caf file dont need to contain that
https://docs.cryengine.com/display/SDKDOC2/Animation+Compression+-+Character+Tool
https://docs.cryengine.com/display/CEMANUAL/Asset+Types
